### ISSUE TITLE
Selecting members of sub arrays results in split objects.

### ISSUE THREAD

**Issue by gak10100** [kofrasa/mingo#106]
Code
```
mingo.find(
  [
    {
      history: [
        {
            "user" : "Jeff",
            "notes" : "asdf"
        }, {
            "user" : "Gary",
        }
      ]
    }
  ],
  {},
  {
    'history.user' : 1,
    'history.notes' : 1
  }
).all()
```
Result
```
[
  {
    history: [
      {
          "user" : "Jeff",
      }, {
          "user" : "Gary",
      }, {
          "notes" : "asdf"
      }
    ]
  }
]
```

Expected Result
```
[
  {
    history: [
      {
          "user" : "Jeff",
          "notes" : "asdf"
      }, {
          "user" : "Gary",
      }
    ]
  }
]
```

**Comment by kofrasa** [comment]
Thanks for the issue, you beat me to creating it. This is another regression not caught by the tests and is similar to #93. 

It appears the bug is sensitive to the order in which the keys are processed when there are missing keys in the nested objects. This can be confirmed by flipping the order of the keys in your example which produces the expected result. 

wrong result for `{ 'history.user' : 1, 'history.notes' : 1 }`
https://runkit.com/embed/2sj75mk6m0zi

correct result for `{ 'history.notes' : 1, 'history.user' : 1 }`
https://runkit.com/embed/rj460rpe9ztx

In the last fix, I added a [test](https://github.com/kofrasa/mingo/commit/ebdc8c0df92550a7a29825e9ed77a093ec5dd825#diff-c5444922198813f5d4a0f082ab84bbb2R464) for missing keys which passed due to the order of the keys.

I noticed this problem after the last release and have since come up with a more robust solution to address it. It should be out soon.
